lib/ex_docker_build.ex

```diff
@@ -201,6 +201,58 @@ defmodule ExDockerBuild do
     end
   end
 
+  @spec get_volumes() :: {:ok, any()} | {:error, any()}
+  def get_volumes do
+    get_volumes(%{})
+  end
+
+  @spec get_volumes(map()) :: {:ok, any()} | {:error, any()}
+  def get_volumes(filters \\ %{}) do
+    case DockerRemoteAPI.get_volumes(filters) do
+      {:ok, %{status_code: 200, body: body}} ->
+        {:ok, Poison.decode!(body)}
+
+      {:ok, %{body: body, status_code: _}} ->
+        {:error, Poison.decode!(body)}
+
+      {:error, %{status_code: 500, message: message}} ->
+      {:error, message}
+    end
+  end
+
+  @spec inspect_volume(String.t()) :: {:ok, any()} | {:error, any()}
+  def inspect_volume(volume_name) do
+    case DockerRemoteAPI.inspect_volume(volume_name) do
+      {:ok, %{status_code: 200, body: body}} ->
+        {:ok, Poison.decode!(body)}
+
+      {:ok, %{body: body, status_code: _}} ->
+        {:error, body}
+
+      {:error, %{reason: reason}} ->
+        {:error, reason}
+    end
+  end
+
+  @spec delete_volume(String.t()) :: :ok | {:error, any()}
+  def delete_volume(volume_name) do
+    case inspect_volume(volume_name) do
```

Comment:
i don't see why we need to inspect a volume before deleting it, we can just try to delete and just it, if there is a failure the user should deal with it